app/core/store/modules/products.js

```diff
@@ -0,0 +1,63 @@
+import Products from 'app/collections/Products'
+
+/**
+ * This module acts as a simple proxy to the Products collection.  Ideally we'd move the products
+ * logic to this module but for now it allows the products module to be called via Vue components.
+ */
+export default {
+  namespaced: true,
+
+  state: {
+    loading: {
+      products: false
+    },
+
+    products: []
+  },
+
+  mutations: {
+    toggleLoadingProducts: (state) => {
+      state.loading.products = !state.loading.products
+    },
+
+    addProducts (state, { products }) {
+      state.products = products
+    }
+  },
+
+  actions: {
+    loadProducts ({ commit }) {
+      commit('toggleLoadingProducts')
+
+      return new Products()
+        .fetch()
+        .done(products => commit('addProducts', { products }))
```

Comment:
For my own understanding. Is the `done`, `fail`, and `always` hooks something the CocoCollections model exposes?

Reply:
Yup, I was hoping it was a promise but unfortunately it's a jqXHR object